packages/ckeditor5-utils/src/keyboard.ts

```diff
@@ -264,9 +282,19 @@ function generateKnownKeyCodes(): { readonly [ keyCode: string ]: number } {
 	}
 
 	// other characters
-	for ( const char of '`-=[];\',./\\' ) {
-		keyCodes[ char ] = char.charCodeAt( 0 );
-	}
+	Object.assign( keyCodes, {
+		'\'': 222,
+		',': 108,
+		'-': 109,
+		'.': 110,
+		'/': 111,
+		';': 186,
+		'=': 187,
+		'[': 219,
+		'\\': 220,
+		']': 221,
+		'`': 223
+	} );
```

Comment:
I'm leaving a comment for the record because this is quite interesting. 

The former implementation was wrong and we hadn't noticed it for years ([not only us](https://stackoverflow.com/questions/27861696/javascript-keycode-clash-right-arrow-and-single-quote)). The thing is, a DOM event's `keyCode` not always aligns with the ASCII table. Sometimes it does ("a", "1", etc.) but for characters like "'" `"'".charCodeAt( 0 )` returns 39 while the actual keystroke code would be 222. 

Because of that, our exported [`keyCodes`](https://ckeditor.com/docs/ckeditor5/latest/api/module_utils_keyboard.html#constant-keyCodes) had duplicated key values ("'" was 39, and "arrowright" was 39), which then became a problem for `keyCodeNames` used by [`getEnvKeystrokeText()`](https://ckeditor.com/docs/ckeditor5/latest/api/module_utils_keyboard.html#function-getEnvKeystrokeText) that reverses the mapping. For instance, as a result, 39 in `keyCodeNames` pointed to "'". So `getEnvKeystrokeText( 'arrowright' )` returned "'". That looked funny when I printed some keystrokes in the `AccessibilityHelp` dialog. ↑, ←, and ↓ looked fine but instead of → there was "'" 😅

We failed to notice this for so long because AFAIR we've never used any of these characters for any feature's keystroke. They wouldn't work because `KeystrokeHandler.set()` also uses [`keyCodes`](https://ckeditor.com/docs/ckeditor5/latest/api/module_utils_keyboard.html#constant-keyCodes) and that would be obvious.

Reply:
A side comment: we should really drop [`keyCode`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode). It's officially deprecated and there are better ways of discovering keys in events. I suppose that would be a BC because `KeystrokeHandler.set()` and `ButtonView#keystroke` format would change.